packages/logger/README.md

```diff
@@ -1,4 +1,320 @@
+const key = 'logger';
+
+export class LoggerComponent implements Component {
+  providers = {
+    [key]: LoggerProvider,  // [key] is a computed property name
+  }
+}
+```
+  Note: What's computed property name ?  See http://es6-features.org/#ComputedPropertyNames
+```
+import {Provider} from '@loopback/context';
+
+export class LoggerProvider implements Provider<SimpleLogger> {
+
+  ... logger implementation comes here.
+
+}
+```
+
+### Custom Sequence
+
+Controllers implement end points and business logic for each end point of the application.   Sequence defines the functional structure of the application.  There can be many end points associated with the application.  There is only one sequence per application.
+
+Usually, the default sequence implemented by LoopBack Next core works for many applications.  In those cases, there is nothing the application needs to do.  To add a piece of new functional element to your application, you will implement the entire sequence in your application.
+
+Client:
+```
+const app = new Application({
+  components: [AuthenticationComponent, LoggerComponent],
+});
+app.sequence(MySequence);
+
+const logger = app.getSync('logger');
+logger.info('My application has started.');
+
+```
+Below is an example implementation of sequence.  We've studied @inject as a parameter decorator.  The four `sequence.actions` keys are bound to the corresponding essential sequence actions.  Since each of the essential sequence actions are implemented by LoopBack Next and in most cases, you can simply inject them in your custom sequence.
+
+The custom sequence below is defined to do something special with req and res objects.  That's the sole purpose of MySequence.
+```
+import {DefaultSequence, FindRoute, InvokeMethod, Reject, Send, inject} from '@loopback/core';
+
+class MySequence extends DefaultSequence {
+  constructor(
+    @inject('sequence.actions.findRoute') protected findRoute: FindRoute,
+    @inject('sequence.actions.invokeMethod') protected invoke: InvokeMethod,
+    @inject('sequence.actions.send') public send: Send,
+    @inject('sequence.actions.reject') public reject: Reject,
+  ) {
+    super(findRoute, invoke, send, reject);
+  }
+
+  async handle(req: ParsedRequest, res: http.ServerResponse) {
+
+    ... we can grab req and res and do something here.  This is the purpose
+    ... of implementing MySequence.
+
+    await super.handle(req, res);
+  }
+}
+
+```
+
+
+# Bonus
+
+### How to test/validate API Spec
+
+As we saw in early part of this blog, the API Spec is attached to the application controller.
+Once attached, the loaded API spec can be accessed by `app.getApiSpec()`
+
+LoopBack Next provides the OpewnAPI spec validator, validateApiSpec as part of the lestlab.
+
+```
+const validateApiSpec = require('@loopback/testlab').validateApiSpec;
+
+describe('validate Api Spec', () => {
+  let app = createApp(); // instantiates the app, its controllers, and the API spec.
+  let apiSpec = app.getApiSpec();
```

Comment:
Test setup should go to a `before` or `beforeEach` hook.

Also test names should read as English sentence, where `it()` stands for the title from `describe`. In this code, we have two sentences: "it apisSpec is valid" (short test name) and "validate Api Spec apisSpec is valid" (full test name), none of read well to me.

```js
describe('Application\'s Api Spec', () => {
  let app, apiSpec;
  before(() => {
    app = createApp();
    apiSpec = app.getApiSpec();
  });

  it('is valid', async () => {
    await validateApiSpec(apiSpec);
  });
});
```